_analyzers/tokenizers/edge-n-gram.md

```diff
@@ -0,0 +1,200 @@
+      "position": 0
+    },
+    {
+      "token": "Code",
+      "start_offset": 0,
+      "end_offset": 4,
+      "type": "word",
+      "position": 1
+    },
+    {
+      "token": "roc",
+      "start_offset": 8,
+      "end_offset": 11,
+      "type": "word",
+      "position": 2
+    },
+    {
+      "token": "rock",
+      "start_offset": 8,
+      "end_offset": 12,
+      "type": "word",
+      "position": 3
+    },
+    {
+      "token": "rocks",
+      "start_offset": 8,
+      "end_offset": 13,
+      "type": "word",
+      "position": 4
+    }
+  ]
+}
+```
+
+## Parameters
+
+| Parameter           | Required/Optional | Data Type        | Description    |
+|:-------|:--------|:------|:---|
+| `min_gram`          | Optional          | Integer          | The minimum token length. Default is `1`.                                                    |
+| `max_gram`          | Optional          | Integer          | The maximum token length. Default is `2`.                                                    |
+| `custom_token_chars`| Optional          | String           | Defines custom characters to be treated as part of a token (for example, `+-_`).                    |
+| `token_chars`       | Optional          | Array of strings | Defines character classes to include in tokens. Tokens are split on characters not in these classes. Default includes all characters. Available classes include: <br> - `letter`: Alphabetic characters (for example, `a`, `ç`, or `京`) <br> - `digit`: Numeric characters (for example, `3`, `7`) <br>- `punctuation`: Punctuation symbols (for example, `!` or `?`) <br> - `symbol`: Other symbols (for example, `$`, `√`) <br> - `whitespace`: Space or newline characters <br> - `custom`: Allows you to specify custom characters in the `custom_token_chars` setting. |
+
+
+## The max_gram parameter limitations
+
+The `max_gram` parameter sets the maximum length of tokens generated by the tokenizer. When a search query exceeds this length, it may fail to match any terms in the index.
+
+For instance, if `max_gram` is set to `4`, the query `explore` would be tokenized as `expl` during indexing. As a result, a search for the full term `explore` will not match the indexed token `expl`.
+
+To address this limitation, you can apply a `truncate` token filter to shorten search terms to the maximum token length. However, this approach has trade-offs. Truncating `explore` to `expl` might lead to matches with unrelated terms like `explosion` or `explicit`, reducing search precision.
+
+We recommend carefully balancing the `max_gram` value to ensure efficient tokenization while minimizing irrelevant matches. If precision is critical, consider alternative strategies, such as adjusting query analyzers or fine-tuning filters.
+
+## Best practices
+
+We recommend using the `edge_ngram` tokenizer only at indexing time to ensure partial word tokens are stored. At search time, a simpler analyzer should be used to match full user queries.
+
```

Comment:
Line 124: The last sentence is a bit too vague. Can we be more precise than "a simpler analyzer" (this is presumably somewhat subjective)? What precisely do we mean by "full user queries"?

Reply:
reworded.